### PR TITLE
UpdateDatamodel 20170703

### PR DESCRIPTION
**Pull request by sjib**
03_qgep_db_dss und 09_qgep_dictionaries

- added missing relations https://github.com/QGEP/QGEP/issues/#284
- Added ON UPDATE CASCADE https://github.com/QGEP/QGEP/issues/#335
- Added z coordinate for reach.progression and wastewater_structure.detailed_geometry_geometry https://github.com/QGEP/QGEP/issues/#331  / https://github.com/QGEP/QGEP/issues/#232
- shortened classnames that give too long primary key constraints https://github.com/QGEP/QGEP/issues/#304
- changed shortcuts that conflict with pgAdmin (IS, OF) https://github.com/QGEP/QGEP/issues/#320
- unique names and shortcuts https://github.com/QGEP/QGEP/issues/#180 (except it / ro

### CODE REVIEW

**Comment by sjib** [comment]
@m-kuhn Warum falle ich bei den Tests durch?

**Comment by sjib** [comment]
Should inlucde the following issues:
#335
#331 3D coordinates
#321
#320
#304
#284
#232

**Comment by m-kuhn** [comment]
Not sure why,
that's what travis prints:

> psql:/home/travis/build/QGEP/datamodel/scripts/../03_qgep_db_dss.sql:3329: ERROR:  column "fk_wastewater_structure" of relation "re_maintenance_event_wastewater_structure" already exists


https://travis-ci.org/QGEP/datamodel/builds/249756347#L205

**Comment by sjib** [comment]
psql:/home/travis/build/QGEP/datamodel/scripts/../03_qgep_db_dss.sql:4535: ERROR:  value too long for type character varying(50)

File: 03_qgep_db_dss.sql [hunk 1]
```diff
@@ -36,7 +36,7 @@ vsacode integer NOT NULL,
 value_en character varying(50),
 value_de character varying(50),
 value_fr character varying(50),
-value_it character varying(50),
+value_it character varying(60),
```
Comment (m-kuhn) [inline 03_qgep_db_dss.sql]: I would propose to keep all languages the same length.
Possibly also just drop the maximum length for less troubles.

**Comment by sjib** [comment]
psql:/home/travis/build/QGEP/datamodel/scripts/../09_qgep_dictionaries.sql:44: ERROR:  duplicate key value violates unique constraint "unq_qgep_is_dictonary_shortcut_it"
DETAIL:  Key (shortcut_it)=(  ) already exists.

**Comment by sjib** [comment]
Solving https://github.com/QGEP/QGEP/issues/180

**Comment by sjib** [comment]
psql:/home/travis/build/QGEP/datamodel/scripts/../09_qgep_dictionaries.sql:44: ERROR:  duplicate key value violates unique constraint "unq_qgep_is_dictonary_shortcut_it"
DETAIL:  Key (shortcut_it)=(NULL) already exists.

**Comment by sjib** [comment]
commented out 
> --ALTER

 TABLE qgep.is_dictionary_od_table ADD CONSTRAINT unq_qgep_is_dictonary_name_it UNIQUE (name_it);
--ALTER TABLE qgep.is_dictionary_od_table ADD CONSTRAINT unq_qgep_is_dictonary_shortcut_it UNIQUE (shortcut_it);
--ALTER TABLE qgep.is_dictionary_od_table ADD CONSTRAINT unq_qgep_is_dictonary_name_ro UNIQUE (name_ro);
--ALTER TABLE qgep.is_dictionary_od_table ADD CONSTRAINT unq_qgep_is_dictonary_shortcut_ro UNIQUE (shortcut_ro);

**Pull request merged**